### ISSUE TITLE
songs I downloaded from the internet will not appear in the playlist immediately

### ISSUE THREAD

**Issue by androrz** [marverenic/Jockey#194]
If I download a song from the internet , and save to sdcard, when I open Jockey again, it was not in the playlist.

**Comment by Dirkstaal** [comment]
You have to refresh the library

**Comment by marverenic** [comment]
Jockey pulls its library data straight from Android using the `MediaStore` database whenever you do a cold start (i.e. remove Jockey from your recent apps and restart it), or by manually refreshing the library. If a song isn't appearing in Jockey, then that means that Android hasn't scanned it yet and that any other music player reading the library from the `MediaStore` will also be missing that song.

There's a delay between when you save a song and when the system scans it that varies wildly between devices depending on how you download it. Sometimes it takes a few seconds, sometimes a few minutes, other times it takes a full reboot. It is possible to force a rescan of the entire SD card, but this is a very expensive process since it involves looking at every single file on the SD card.

If it's only a minor delay between when it appears in Jockey's library and it's consistent with all other music players, then it's not an issue that can reasonably be fixed. If it never shows up in Jockey but it does show up in another music player, then I'll need more information like how you're downloading songs, what other music player shows the song, how long the delay is (roughly), and what phone and Android version you're using.

**Comment by Khang-NT** [comment]
Is Jockey auto refresh library after MediaStore has changed? Some like:
```java
getContentResolver().registerContentObserver(
                android.provider.MediaStore.Audio.Media.EXTERNAL_CONTENT_URI,
                true,
                theObserver /** do refresh library on changed **/);
```

**Comment by marverenic** [comment]
Automatic refresh is implemented in #255.

**Comment by androrz** [comment]
this is not a bug, i was testing demo flavor
